### PR TITLE
feat: add workflow.labels and workflow.annotations as JSON string. Fixes: #7289

### PR DESCRIPTION
**Pull request by NikeNano**
Signed-off-by: NikeNano <niklas.sven.hansson@gmail.com>

Don't bother creating a PR until you've done this:

* [ ] Run `make pre-commit -B` to fix codegen, lint, and commit message problems.

Create your PR as a draft.

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it
  does not need to pass.
* Once required tests have passed, you can make it "Ready for review".
* Say how how you tested your changes. If you changed the UI, attach screenshots.

Tips:

* If changes were requested, and you've made them, then dismiss the review to get it looked at again.
* Add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md) if you like.
* You can ask for help!

### CODE REVIEW

File: workflow/controller/operator.go [hunk 1]
```diff
@@ -539,9 +539,15 @@ func (woc *wfOperationCtx) setGlobalParameters(executionParameters wfv1.Argument
 			return fmt.Errorf("either value or valueFrom must be specified in order to set global parameter %s", param.Name)
 		}
 	}
+	if workflowAnnotations, err := json.Marshal(woc.wf.ObjectMeta.Annotations); err == nil {
+		woc.globalParams[common.GlobalVarWorkflowAnnotations] = string(workflowAnnotations)
+	}
```
Comment (simster7) [inline workflow/controller/operator.go:544]: Will these be updated every operation?

**Comment by sarabala1979** [comment]
@NikeNano There is duplicate PR for this feature. Another PR is ready to review.  But your PR was created before him. I would like to give preference to you. Do you have any ETA to finish up this?

**Comment by NikeNano** [comment]
> @NikeNano There is duplicate PR for this feature. Another PR is ready to review. But your PR was created before him. I would like to give preference to you. Do you have any ETA to finish up this?

I can make the updates tomorrow(16 December), to push it forward.

File: workflow/controller/operator.go (hunk 1 above)
Comment (NikeNano) [reply workflow/controller/operator.go:544]: I assume the annotations and the labels will not be updated for every operation. What do you suggest here?

**Comment by NikeNano** [comment]
Took the validation logic that you had added @dpadhiar and added you as co-author hope that is fine with you, if not let me know.

**Comment by dpadhiar** [comment]
> Took the validation logic that you had added @dpadhiar and added you as co-author hope that is fine with you, if not let me know. Happy to resolve this together @dpadhiar

Totally fine! Happy to resolve the issue with you as well.

**Review by sarabala1979 — APPROVED** [review]

**Pull request merged**